libraries/helpers.rb

```diff
@@ -96,24 +96,24 @@ def product_lookup(product, version = '0.0.0')
       # implying latest, we need to additionally ensure that the
       # bottom version is something valid. If we don't have the check
       # in the elsif, it will say that 0.0.0 is not a valid version.
-      if (product == 'chef-server')
+      if product == 'chef-server'
         if (v < Mixlib::Versioning.parse('12.0.0')) && (v > Mixlib::Versioning.parse('11.0.0'))
           data['package-name'] = 'chef-server'
-        elsif (v < Mixlib::Versioning.parse('11.0.0')) && (v > Mixlib::Versioning.parse('1.0.0'))
+        elsif (v < Mixlib::Versioning.parse('11.0.0')) && (v > Mixlib::Versioning.parse('0.0.0'))
           Chef::Log.fatal("Invalid version specified, '#{version}' for #{product}!")
           fail
         end
-      elsif (product == 'manage') && (v < Mixlib::Versioning.parse('2.0.0'))
+      elsif (product == 'manage') && ((v < Mixlib::Versioning.parse('2.0.0')) && (v > Mixlib::Versioning.parse('0.0.0')))
         data['package-name'] = 'opscode-manage'
         data['ctl-command'] = 'opscode-manage-ctl'
       # TODO: When Chef Push server and client 2.0 are released, we
       # need to implement similar logic to chef-server, so that the
       # default "latest" version, 0.0.0 (no constraint) doesn't result
       # in the old package.
-      elsif (product == 'push-server') && (v < Mixlib::Versioning.parse('2.0.0'))
+      elsif (product == 'push-server') && ((v < Mixlib::Versioning.parse('2.0.0')) && (v > Mixlib::Versioning.parse('0.0.0')))
```

Comment:
Sure, sorry I wasn't more specific: 

```
* apt_package[manage] action install[2015-11-21T18:03:58+00:00] INFO: Processing apt_package[manage] action install (/var/chef/cache/cookbooks/chef-ingredient/libraries/debian_handler.rb line 66)
                        
  * No candidate version available for chef-manage
  ================================================================================
  Error executing action `install` on resource 'apt_package[manage]'
  ================================================================================
  
  Chef::Exceptions::Package
  -------------------------
  No candidate version available for chef-manage
  
  Resource Declaration:
  ---------------------
  # In /var/chef/cache/cookbooks/chef-ingredient/libraries/debian_handler.rb
  
   66:         package new_resource.product_name do # ~FC009
   67:           action action_name
   68:           package_name ingredient_package_name
   69:           options new_resource.options
   70:           timeout new_resource.timeout
   71: 
   72:           # If the user specifies "0.0.0", :latest or "latest" as version,
   73:           # we should not give any version to the package resource.
   74:           if Mixlib::Versioning.parse(version_string(new_resource.version)) > '0.0.0'
   75:             version version_for_package_resource
   76:           end
   77: 
  
  Compiled Resource:
  ------------------
  # Declared in /var/chef/cache/cookbooks/chef-ingredient/libraries/debian_handler.rb:66:in `configure_package'
  
  apt_package("manage") do
    action [:install]
    retries 0
    retry_delay 2
    default_guard_interpreter :default
    package_name "chef-manage"
    declared_type :package
    cookbook_name "chef-server-12"
  end
```

Lemme know if you need any additional details.